Carrot-Assistant/components/models.py

```diff
@@ -49,39 +139,565 @@ def get_model(
     object
         An interface to generate text using an LLM
     """
-    if "gpt" in model_name.lower():
-        logger.info(f"Loading {model_name} model")
+    # Normalize the huggingface for downloading the model
+    from huggingface_hub import hf_hub_download
 
-        llm = OpenAIGenerator(
-            model=model_name, generation_kwargs={"temperature": temperature}
-        )
+    # Normalize the model name to lowercase
+    model_name = model_name.lower()
 
-    else:
-        logger.info(f"Loading {model_name} model")
-        from huggingface_hub import hf_hub_download
+    # Select device based on GPU availability
+    device = -1 if torch.cuda.is_available() else 0
 
-        device = -1 if torch.cuda.is_available() else 0
+    try:
+        # ------ GPT Model ------ >
+        # --------------------------
 
-        try:
-            llm = LlamaCppGenerator(
-                model=hf_hub_download(**local_models[model_name]),
-                n_ctx=0,  # Text context, 0 = from model
-                n_batch=512,
-                model_kwargs={"n_gpu_layers": device, "verbose": True},
-                generation_kwargs={"max_tokens": 128, "temperature": temperature},
-            )
-        except KeyError:
-            print(f"{model_name} is not a recognised model name")
-        except:
-            print(f"Error loading {model_name}")
-        finally:
-            logger.info("Loading llama-3.1-8b")
-            llm = LlamaCppGenerator(
-                    model=hf_hub_download(**local_models[model_name]),
+        if "gpt" in model_name:
+            logger.info(f"Loading {model_name} from OpenAI API")
+            try:
+                llm = OpenAIGenerator(
+                    model=model_name, generation_kwargs={"temperature": temperature}
+                )
+            except Exception as e:
+                logger.error(f"Failed to load GPT-based model {model_name}: {e}")
+                raise ValueError(f"Failed to load GPT-based model {model_name}: {e}")
+
+        # ------ Llama Model (7B) ------ >
+        # --------------------------
+
+        elif model_name == "llama2-7b":
```

Comment:
It would be less verbose to use hf_hub_download(**local_models[model_name]) than this elif chain

Reply:
Sure